executor/executor_write.go

```diff
@@ -407,6 +411,180 @@ func (e *DeleteExec) Close() error {
 	return e.SelectExec.Close()
 }
 
+// NewLoadDataInfo returns a LoadDataInfo structure, and it's only used for tests now.
+func NewLoadDataInfo(row []types.Datum, ctx context.Context, tbl table.Table) *LoadDataInfo {
+	return &LoadDataInfo{
+		row:       row,
+		insertVal: &InsertValues{ctx: ctx, Table: tbl},
+		Table:     tbl,
+	}
+}
+
+// LoadDataInfo saves the information of loading data operation.
+type LoadDataInfo struct {
+	row       []types.Datum
+	insertVal *InsertValues
+
+	Path       string
+	Table      table.Table
+	FieldsInfo *ast.FieldsClause
+	LinesInfo  *ast.LinesClause
+}
+
+// InsertData inserts data into specified table according to the specified format.
+// If it has the rest of data isn't completed the processing, then is returns without completed data.
+// If prevData isn't nil and curData is nil, there are no other data to deal with and the isEOF is true.
+func (e *LoadDataInfo) InsertData(prevData, curData []byte) ([]byte, error) {
+	// TODO: support enclosed and escape.
+	var isEOF bool
+	var notLineEnd bool
+	cols := make([]string, 0, len(e.row))
+	startingLen := len(e.LinesInfo.Starting)
+	terminatedLen := len(e.LinesInfo.Terminated)
+	if len(prevData) > 0 && len(curData) == 0 {
+		isEOF = true
+		prevData, curData = curData, prevData
+	}
+	for len(curData) > 0 {
+		endIdx := strings.Index(string(curData), e.LinesInfo.Terminated)
+		if endIdx == -1 {
+			if len(prevData) > 0 {
+				curData = append(prevData, curData...)
+			}
+			endIdx = len(curData)
+			notLineEnd = true
```

Comment:
We can check if Terminated can be found after we have append prevData and curData, if not found, we can return the appended data directly, then we only need to handle the case when there is terminate string.